### DISCUSSION TITLE
Any solutions to capture effective sycl AST information

### DISCUSSION THREAD

**Discussion by malixian** [intel/llvm#5445] [Q&A] [answered]
I use the command `clang++ -Xclang -ast-dump -fsycl -fsycl-device-only vector_add.cpp` to dump dpc++ AST. And  I get a more than 100'000 lines output.
It seems difficult to process this AST, so is there any solution to capture effective sycl AST information?

**Comment by Fznamznon** [comment, marked as answer]
AFAIK there is no other official way to examine AST coming from clang. The AST dump you are seeing is quite big because `-ast-dump` shows it for the whole translation unit which becomes quite big together with all included standard C++ and DPC++ headers.

It is possible to hack locally built clang by adding calls to `dump()` method of concrete AST nodes, but it requires some knowledge of codebase and it can be hard to achieve concrete set of dumped nodes, so please don't see it as like I'm suggesting this as a solution.
